runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -0,0 +1,279 @@
+            openReplicationStatusTable();
+        }
+
+        private void openReplicationStatusTable() {
+            try {
+                corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, REPLICATION_STATUS_TABLE, ReplicationStatusKey.class,
+                    ReplicationStatusVal.class, null, TableOptions.fromProtoSchema(ReplicationStatusVal.class));
+            } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+                log.error("Failed to open the replication status table", e);
+                throw new UnrecoverableCorfuError(e);
+            }
+        }
+
+        @Override
+        public void run() {
+            CorfuStoreMetadata.Timestamp timestamp = getValidSubscriptionTimestamp();
+
+            if (bufferSize == DEFAULT_BUFFER_SIZE) {
+                corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeAcrossNamespaces(clientListener,
+                        namespace, streamTag, tablesOfInterest, timestamp.getSequence());
+            } else {
+                corfuStore.getRuntime().getTableRegistry().getStreamingManager().subscribeAcrossNamespaces(clientListener,
+                        namespace, streamTag, tablesOfInterest, timestamp.getSequence(), bufferSize);
+            }
+        }
+
+        /**
+         * Returns a valid timestamp at which the client stream listener can be subscribed.  A valid timestamp is one
+         * which is outside the bounds of an ongoing full-sync.   Client callbacks are also invoked to ensure that
+         * updates upto this timestamp have been read and processed.
+         */
+        private CorfuStoreMetadata.Timestamp getValidSubscriptionTimestamp() {
+
+            // Number of milliseconds to wait before each retry
+            int backoffRetryDurationMs = 5;
+
+            // Max duration at which the wait time must be capped (2 minutes)
+            int maxDurationThresholdMins = 2;
+
+            // Retry until a timeout of max_duration_threshold_mins
+            try {
+                return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                    try {
+                        // Check if snapshot sync is currently in progress.  If in progress, wait for it to finish
+                        CorfuStoreMetadata.Timestamp latestTimestamp = getTimestamp();
+                        if (checkSnapshotSyncOngoing(latestTimestamp)) {
+                            waitSnapshotSyncCompletion(latestTimestamp);
+                        }
+
+                        // Invoke the client callback for multi table reads.  This will read all tables the client is
+                        // interested in and return a timestamp.  Verify if snapshot sync was in progress at that
+                        // timestamp.  If in progress, wait for it to complete and retry the whole workflow.
+                        CorfuStoreMetadata.Timestamp multiTableReadTimestamp = clientListener.performMultiTableReads();
```

Comment:
I think I even had something to do with this name, but I'm not completely convinced about it. Because for the clients perspective its a full sync, too seems more appropriate fullSync()?

Reply:
My thought process is that (1) Full Sync is the client' common terminology (which differs from snapshot sync used in LR) and (2) the signature indicates we are not providing anything (no snapshot data) so it's their callback to do a full sync for all their tables to merge.